Add advice about number of nodes to distributed_deployment (#692)
qdrant/landing_page: qdrant-landing/content/documentation/guides/distributed_deployment.md

Since version v0.8.0 Qdrant supports a distributed deployment mode.
In this mode, multiple Qdrant services communicate with each other to distribute the data across the peers to extend the storage capabilities and increase stability.

> ## How many Qdrant nodes should I run?

Reviewer: Would it make sense to organize this in a table?

Alternatively, if a table is too complicated, maybe set up bullets, with highlighted values, such as:

```
The number of Qdrant nodes you want depends on your priorities:

- **Costs**: If costs are most important, run a single Qdrant node. Drawbacks:
  - Downtime: Users will experience downtime during node restarts.
  - Performance: Limited to the resources of a single server.
  - Resilience: Recovery is not possible unless you have backups or snapshots.

- **Less downtime**: Run two Qdrant nodes with replicated shards. Drawbacks:
   - The cluster still responds to most read/write requests even when one node fails.
   - The cluster cannot run operations on collections.
   - Increased performance/scalability. You need not replicate shards to every node.
   - Increased performance with less RAM or disk space. However, this comes at a cost: resilience suffers, since non-replicated shards cannot be served when the node holding that shard is down.
   - Since collection operations are rarely needed, two-node clusters are a good balance between resilience and cost.

- **Best performance/scalability**: Three or more Qdrant nodes.
   - Assumption: Shards are replicated to at least two nodes. 
   - All operations continue to work when one node is down
   - Increased performance/scalability from multiple nodes.
   
```

}
```

## Enabling distributed mode in Qdrant Cloud

> In Qdrant Cloud, when you click "Scale Up" to increase your cluster size to >1, the distributed mode settings are set automatically, resulting in a new empty node starting up. For best results, ensure your cluster is running Qdrant v1.7.4 or higher.

Reviewer: I've made some assumptions here:

```suggestion
In Qdrant Cloud, when you "Scale Up" to increase set up more than one cluster, the distributed mode settings are set automatically. A new empty node starts up. For best results, ensure your cluster is running Qdrant v1.7.4 or higher.
```

Author: Will make this clearer by explicitly pointing the user to the next section 👍

## Making use of a new distributed Qdrant cluster

> When you enable distributed mode and scale up to 2 or more nodes, your data does not move to the new node automatically; it starts out empty. To make use of your new empty node, you will have to do one of the following:
> 
> * You can replicate your existing data to the new node by [creating new shard replicas](#creating-new-shard-replicas)
> * You can create a new replicated collection by setting the [replication_factor](#replication-factor) to 2 or more (can only be set at creation time)
> * You can move data (without replicating it) onto the new node by [moving shards](#moving-shards)

Reviewer: ```suggestion
When you enable distributed mode and scale up to 2 or more nodes, your data does not move to the new node automatically; it starts out empty. To make use of your new empty node, do one of the following:

* Replicate your existing data to the new node by [creating new shard replicas](#creating-new-shard-replicas)
* Create a new replicated collection by setting the [replication_factor](#replication-factor) to 2 or more (can only be set at creation time)
* Move data (without replicating it) onto the new node by [moving shards](#moving-shards)
```

Once all shards of the collection are recovered, the collection will become operational again.

### Temporary node failure

> If properly configured, running Qdrant in distributed mode can make your cluster resistent to outages when one node is down temporarily.

Reviewer: Does this still work in a "one-node" configuration? If not:

```suggestion
If you have a multi-node configuration, you can run Qdrant in distributed mode. This can make your cluster resistant to outages when one node fails.
```

> If properly configured, running Qdrant in distributed mode can make your cluster resistent to outages when one node is down temporarily.
> 
> Here is how differently-configured Qdrant clusters respond to one node going down temporarily:
> 
> * 1-node clusters: All operations will time out or fail for a few second to a few minutes, depending on how long it takes to restart and load data from disk.
> * 2-node clusters where shards ARE NOT replicated: All operations will time out or fail for a few second to a few minutes, depending on how long it takes to restart and load data from disk.
> * 2-node clusters where all shards ARE replicated to both nodes: All requests except for operations on collections will continue to work during the outage.
> * 3+-node clusters where all shards are replicated to at least 2 nodes: All requests will continue to work during the outage.

Reviewer: ```suggestion
If properly configured, running Qdrant in distributed mode can make your cluster resistant to outages when one node fails temporarily.

Here is how differently-configured Qdrant clusters respond:

* 1-node clusters: All operations time out or fail. The failure may last as long as a few minutes. It depends on how long it takes to restart and load data from disk.
* 2-node clusters where shards ARE NOT replicated: All operations time out or fail for up to a few minutes. It depends on how long it takes to restart and load data from disk.
* 2-node clusters where all shards ARE replicated to both nodes: All requests except for operations on collections continue to work during the outage.
* 3+-node clusters where all shards are replicated to at least 2 nodes: All requests will continue to work during the outage.
```
